sessionctx/binloginfo/binloginfo.go

```diff
@@ -57,6 +59,54 @@ func GetPumpsClient() *pumpcli.PumpsClient {
 	return client
 }
 
+// GetOrCreatePumpsClient get or create the pumps client instance
+func GetOrCreatePumpsClient() *pumpcli.PumpsClient {
+	pumpsClientLock.RLock()
+	client := GetPumpsClient()
+	if client == nil {
```

Comment:
if two session call this function at the same time, and all get a nil client, will create pump client twice. 
how about add read lock when call GetPumpsClient, and if client is nil, use lock, and then create pump client.